test/public.spec.js

```diff
@@ -0,0 +1,423 @@
+      });
+
+      it('should show all info', function() {
+        let textLines = getImageOverlayTextLines(0);
+
+        expect(textLines).toContain(jasmine.stringMatching('...holder-100x80.png'));
+        expect(textLines).toContain('Display: 200 x 160');
+        expect(textLines).toContain('Natural: 100 x 80');
+        expect(textLines).toContain('Image coverage: 25.00%');
+        expect(textLines).toContain('File Size: 4.464 KB');
+      });
+
+      it('should have url link to new tab', function() {
+        let imageOverlay = getImageOverlay(0);
+        let urlLink = imageOverlay.getElementsByTagName('a')[0];
+
+        expect(urlLink.href).toEqual(jasmine.stringMatching('test/assets/placeholder-100x80.png'));
+        expect(urlLink.target).toEqual('_blank');
+        expect(urlLink.innerText).toEqual(jasmine.stringMatching('...holder-100x80.png'));
+      });
+    });
+
+    describe('medium images overlays', function() {
+      beforeEach(function() {
+        createDomNodes([
+          createMediumImg()
+        ]);
+        window.NCC.imageChecker.showImagesInfo();
+      });
+
+      it('should not show url info', function() {
+        let imageOverlay = getImageOverlay(0);
+        let urlLink = imageOverlay.getElementsByTagName('a')[0];
+
+        expect(urlLink).toBeUndefined();
+      });
+
+      it('should show other info', function() {
+        let textLines = getImageOverlayTextLines(0);
+
+        expect(textLines).toContain('Display: 151 x 71');
+        expect(textLines).toContain('Natural: 100 x 80');
+        expect(textLines).toContain('Image coverage: 74.62%');
+        expect(textLines).toContain('File Size: 4.464 KB');
+      });
+
+      it('should click to new tab', function() {
+        let urlLink = getImageOverlay(0);
+
+        expect(urlLink.href).toEqual(jasmine.stringMatching('test/assets/placeholder-100x80.png'));
+        expect(urlLink.target).toEqual('_blank');
+      });
+    });
+
+    describe('small images overlays', function() {
+      beforeEach(function() {
+        createDomNodes([
+          createSmallImg()
+        ]);
+        window.NCC.imageChecker.showImagesInfo();
+      });
+
+      it('should only have title', function() {
+        let imageOverlay = getImageOverlay(0);
+        let titleParts = imageOverlay.title.split(',');
+        expect(titleParts[0].trim()).toEqual('Coverage: 106.67%');
+        expect(titleParts[1].trim()).toEqual('File Size: 4.464 KB');
+        expect(titleParts[2].trim()).toEqual(jasmine.stringMatching('URL: .*test/assets/placeholder-100x80.png'));
+      });
+
+      it('should not have any content', function() {
+        let textLines = getImageOverlayTextLines(0);
+        expect(textLines).toEqual(['']);
+      });
+    });
+
+    describe('cross domain images overlays', function() {
+      beforeEach(function() {
+
+      });
+
+      it('should not show file size', function(done) {
+        createDomNodes([
```

Comment:
... I'm assuming you can't use `beforeEach` because the load event has already fired.  Can the whole `onload` callback be moved into `beforeEach` instead?